### DISCUSSION TITLE
Revit developer starter pack

### DISCUSSION THREAD

**Discussion by Nice3point** [jeremytammik/RevitLookup#209] [General]
Starting to develop an add-in for Revit can be quite time-consuming, and mastering the process can take a significant amount of effort. To simplify and streamline this journey, I've created a collection of libraries aimed at automating many of the development processes.

# Revit templates

<img width="400" src="https://github.com/Nice3point/RevitTemplates/assets/20504884/ddeb2dd2-e3e9-46f8-a643-4176a09c8560">

https://github.com/Nice3point/RevitTemplates

RevitTemplates offer a variety of templates specifically tailored for Revit add-in development. 
They are compatible with popular IDEs such as Visual Studio, Jetbrains Rider, and CLI as well. 
What's more, they fully support all Revit versions, from 2020 to 2025, within a single solution. 
These templates also support key development practices such as MVVM, Dependency Injection, and Logging. 
One of the highlights is the inclusion of optional features, allowing you to customize your project according to your specific needs. 
For enterprise-level development, the templates even provide a comprehensive build system, installer creation, and a bundle for seamless publishing to platforms like the [Autodesk Store or Design Automation](https://github.com/Nice3point/RevitTemplates/wiki/Autodesk-Store-bundle).

Template description: https://github.com/Nice3point/RevitTemplates/wiki/Templates
Examples created with these templates: https://github.com/Nice3point/RevitTemplates/tree/develop/samples
Revit multi-version support: https://github.com/Nice3point/RevitTemplates/wiki/Multiple-Revit-Versions

# Revit toolkit

<img width="320" src="https://github.com/Nice3point/RevitToolkit/assets/20504884/c59042df-b9b5-4829-9417-006912781cf2">

https://github.com/Nice3point/RevitToolkit

This package offers implementations of interfaces commonly used in Revit development. 
By integrating this toolkit into your project, you can streamline your codebase, reduce infrastructure code and enhance dependency resolution. 
Notably, it includes implementations for IExternalEventHandler, facilitating the creation of modeless applications and the execution in async context.

# Revit extensions

<img width="400" src="https://github.com/Nice3point/RevitExtensions/assets/20504884/a1772d7d-38d4-4a9b-9985-1d83b8cbea8d">

https://github.com/Nice3point/RevitExtensions

RevitExtensions are designed to reduce the burden of writing repetitive code in Revit projects, and also help you write chained methods.
You don't have to write huge code to add a button to a Revit ribbon or remember all Utils classes, everything is available at your fingertips.
Just try.

# Revit API

<img width="250" src="https://github.com/Nice3point/RevitApi/assets/20504884/fdc24b70-1dce-44f9-961f-7c4b467cc024">

https://github.com/Nice3point/RevitApi

The RevitApi package serves as a dependency package for add-in development. 
Instead of relying on local references, it is recommended to use Nuget for correct project dependency restoring and support for continuous integration and continuous deployment (CI/CD).

# RevitLookup

<img alt="RevitLookup" width="320" src="https://github-production-user-asset-6210df.s3.amazonaws.com/20504884/243138821-45a22919-4865-478e-8cfe-dc6288646d44.png">

Lastly, I'm also the developer of RevitLookup, a tool designed to facilitate quick analysis of the Revit database. It's an invaluable resource for anyone working with Revit. 
You can find the latest version of RevitLookup [here](https://github.com/jeremytammik/RevitLookup).

By leveraging these resources, you can significantly enhance your Revit development workflow, saving time and effort while ensuring consistency and efficiency in your projects.

**Comment by MartenKL** [comment]
How do I debug projects created with these templates (Revit 2025)? I set a launch profile to start Revit 2025 but that doesn't load the addin in Revit 2025.

**Reply by Nice3point** [reply]
Just set "Debug R25" configuration and run Debug session 
![изображение](https://github.com/jeremytammik/RevitLookup/assets/20504884/86d30e36-3022-496d-8d87-347f49f3d8ad)

if you have any troubles or have modified your .csproj, please share your solution
